nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/util/CollectionNameValidator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.nifi.processors.mongodb.util;
+
+import org.apache.nifi.components.ValidationContext;
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.processor.util.StandardValidators;
+
+/**
+ * Validator for the collection name field. It is derived from the rules published in the documentation for
+ * MongoDB here:
+ *
+ * https://docs.mongodb.com/manual/reference/limits/#restrictions-on-collection-names
+ */
+public class CollectionNameValidator implements Validator {
+    private static final CollectionNameValidator INSTANCE = new CollectionNameValidator();
+
+    private CollectionNameValidator() {
+
+    }
+
+    @Override
+    public ValidationResult validate(String subject, String input, ValidationContext context) {
+        ValidationResult.Builder builder = new ValidationResult.Builder();
+        builder.subject(subject).input(input);
+
+        ValidationResult elCheck = StandardValidators.NON_EMPTY_EL_VALIDATOR.validate(subject, input, context);
+
+        if (!elCheck.isValid()) {
+            builder.valid(false).explanation(elCheck.getExplanation());
+        } else if (input.contains("$") && !elCheck.isValid()) {
+            builder.valid(false).explanation("Collection name cannot have the \"$\" character in it.");
```

Comment:
I don't think this is working. I guess the intention here is to 
1. check if EL is present and not valid, display the explanation 
2. If EL is not present and the provided value has a `$` in it, display the appropriate explanation.

I gave a value like `te$tdb` and `te$tcollection` for `GetMongo`. Ideally the processor should be in invalid state going by the above rules but it was in ready state. Please correct me if I'm wrong.

Reply:
Ok, I'll take a look.

Reply:
It looks like that is seen as another form of valid expression language. I just added a new unit test that verifies it as closely as I think we can get.